homeassistant/components/osramlightify/light.py

```diff
@@ -4,270 +4,410 @@
+        self._min_mireds = 0
+        self._max_mireds = 0
+        self._brightness = 0
+        self._color_temp = 0
+        self._hs_color = (0, 0)
+        self._rgb_color = (0, 0, 0)
+        self._xy_color = (0, 0)
+
+        self.update_static_attributes()
+        self.update_dynamic_attributes()
+
+    def _get_unique_id(self):
+        """Get a unique ID (not implemented)."""
+        raise NotImplementedError
+
+    def _get_supported_features(self):
+        """Get list of supported features."""
+        features = 0
+        if 'lum' in self._luminary.supported_features():
+            features = features | SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+
+        if 'temp' in self._luminary.supported_features():
+            features = features | SUPPORT_COLOR_TEMP | SUPPORT_TRANSITION
+
+        if 'rgb' in self._luminary.supported_features():
+            features = (features | SUPPORT_COLOR | SUPPORT_TRANSITION |
+                        SUPPORT_EFFECT)
+
+        return features
+
+    def _get_effect_list(self):
+        """Get list of supported effects."""
+        effects = []
+        if 'rgb' in self._luminary.supported_features():
+            effects.append(EFFECT_RANDOM)
+
+        return effects
 
     @property
     def name(self):
-        """Return the name of the device if any."""
-        return self._name
+        """Return the name of the luminary."""
+        return self._luminary.name()
 
     @property
     def hs_color(self):
-        """Last hs color value set."""
-        return self._hs
+        """Return last hs color value set."""
+        return self._hs_color
 
     @property
     def color_temp(self):
         """Return the color temperature."""
-        return self._temperature
+        return self._color_temp
 
     @property
     def brightness(self):
-        """Brightness of this light between 0..255."""
+        """Return brightness of the luminary (0..255)."""
         return self._brightness
 
     @property
     def is_on(self):
-        """Update status to True if device is on."""
-        return self._state
+        """Return True if the device is on."""
+        return self._is_on
 
     @property
     def supported_features(self):
-        """Flag supported features."""
-        return SUPPORT_OSRAMLIGHTIFY
+        """List of supported features."""
+        return self._supported_features
 
     @property
     def effect_list(self):
         """List of supported effects."""
-        return [EFFECT_RANDOM]
+        return self._effect_list
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return self._min_mireds
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        return self._max_mireds
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
+
+    @property
+    def state_attributes(self):
```

Comment:
Delete this method. The core class uses your properties to build the information and you should not copy the method itself.

Reply:
Ok.  You are probably 100% right from the maintainability perspective.  The reason I have re-implemented this method is just to avoid redundant recalculations of ATTR_HS_COLOR, ATTR_RGB_COLOR and ATTR_XY_COLOR during each state reading (each X seconds).  Instead I have moved that logic to update method (each X minutes or even hours).  But CPU footprint should be very small anyway.